src/dotty/tools/dotc/transform/FullParameterization.scala

```diff
@@ -86,9 +88,12 @@ trait FullParameterization {
    *    }
    *
    *  If a self type is present, $this has this self type as its type.
+   *
    *  @param abstractOverClass  if true, include the type parameters of the class in the method's list of type parameters.
+   *  @param liftThisType       if true, require created $this to be $this: (Foo[A] & Foo,this).
```

Comment:
Why not have the type of `$this` simply be `Foo.this` when `liftThisType` is true?